agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/JsonConfigPolling.java

```diff
@@ -46,22 +43,22 @@
 
     private final Path path;
     private volatile long lastModifiedTime;
-    private volatile double lastReadSamplingPercentage;
+    private volatile Configuration lastReadConfig;
     private static final Logger logger = LoggerFactory.getLogger(JsonConfigPolling.class);
 
-    // visible for testing
-    JsonConfigPolling(Path path, long lastModifiedTime, double lastReadSamplingPercentage) {
-        this.path = path;
-        this.lastModifiedTime = lastModifiedTime;
-        this.lastReadSamplingPercentage = lastReadSamplingPercentage;
-    }
-
     // passing in lastReadSamplingPercentage instead of using the real samplingPercentage, because the real
     // samplingPercentage is rounded to nearest 100/N, and we want to know specifically when the underlying config value changes
     // which is lastReadSamplingPercentage
-    public static void pollJsonConfigEveryMinute(Path path, long lastModifiedTime, double lastReadSamplingPercentage) {
+    public static void pollJsonConfigEveryMinute(Path path, long lastModifiedTime, Configuration lastReadConfig) {
```

Comment:
is it necessary to expose whole config? i don't find anything else is used except the sampling rate.